xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-requiredrights/xwiki-platform-security-requiredrights-rest/src/main/java/org/xwiki/platform/security/requiredrights/rest/RequiredRightsRestResource.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.platform.security.requiredrights.rest;
+
+import javax.ws.rs.Encoded;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+import org.xwiki.rest.XWikiRestException;
+import org.xwiki.security.requiredrights.rest.model.jaxb.DocumentRequiredRights;
+import org.xwiki.security.requiredrights.rest.model.jaxb.DocumentRightsAnalysisResult;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Get the result of the required rights analysis of a page.
+ *
+ * @version $Id$
+ * @since 17.4.0RC1
+ */
+@Path("/wikis/{wikiName}/spaces/{spaceName: .+}/pages/{pageName}/requiredRights")
+@Unstable
+public interface RequiredRightsRestResource
+{
+    /**
+     * @param wiki the wiki of the document to get annotations for
+     * @param spaceNames the space names of the document to get annotations for
+     * @param page the name of the document to get annotation for
+     * @return the result of the required rights analysis of a page
+     * @throws XWikiRestException when failing to parse space
+     */
+    @GET
+    DocumentRightsAnalysisResult analyze(@PathParam("spaceName") @Encoded String spaceNames,
```

Comment:
I triggers me a bit to see that a GET request triggers an analysis here... is there a risk it could be used to perform DDOS?

Reply:
The analysis should be less expensive than rendering the page. What we could do is to limit this API to users with edit right as we currently only use this API for users with edit right.

Reply:
It would probably make sense then. Do you see a limitation with doing that?

Reply:
Does not make much sense to me, this API is manipulating things which are accessible to a user with VIEW right on the document, so I find it very strange to limit it to EDIT users.